mautic/mautic
# New form field: Company lookup

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | /

#### Description:
This PR adds a new form field called **Company Lookup** that allows contacts to type and see what companies already exist in the database. If there is a company that they want to add, they can select it from the options. Users cannot create new companies with this field. Only select those that are already companies in Mautic.

This improves the UX for contacts and for users as there will be less company duplicates.
![Screenshot 2024-07-22 at 17 12 56](https://github.com/user-attachments/assets/0744772a-0b82-4dd4-bcff-61e6bfba0567)

![Screenshot 2024-07-22 at 17 12 21](https://github.com/user-attachments/assets/8e6688ac-9ea6-4a49-afff-468c153584f4)

![Screenshot 2024-07-23 at 17 17 09](https://github.com/user-attachments/assets/235d72d2-c402-4075-af13-bba9919c558c)

#### Steps to test this PR:
1. Create a bunch of companies in Mautic.
1. Create a new form or edit an already existing form.
1. Go to fields and add a new field with the type 'Company Lookup'.
1. Set the field configurations as per your requirement.
1. Save the form.
1. Access the form by entering the form URL in a new browser tab (Or any other way supported by Mautic i.e through JS embed or part of the campaign).
1. Verify the working of the lookup field by entering any text value. You would notice that the lookup field automatically fetches hints based on the configurations. Once, the user selects any value, the value is auto-filled in the respective field.


[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. FieldModelTest:: testGetFieldsProperties
2. PublicControllerFunctionalTest
